sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -185,7 +185,11 @@ public override RequestMethod Method
 
             public override RequestContent? Content { get; set; }
 
-            public override string ClientRequestId { get; set; }
+            public override string ClientRequestId
+            {
+                get => _clientRequestId ??= Guid.NewGuid().ToString();
+                set => _clientRequestId = value;
+            }
```

Comment:
Setting `ClientRequestId` to `null` is going to be a little confusing for folks.  Is there a way to disable this policy?  I didn't see one when I looked quickly, but if so we might consider throwing an exception for `null` and directing users to the proper way to disable this header from being set.  I'm also okay with the current behavior.